js/background/background.js

```diff
@@ -736,6 +736,76 @@ class Steam {
 Steam._dynamicstore_promise = null;
 Steam._supportedCurrencies = null;
 
+// When redirecting to Steam Community, if the game does not actually exist, onHeadersReceived would normally trigger an infinite redirection loop. To avoid that, we store the IDs of the requests that have already been redirected here.
+let webRequests = [];
+
+function onHeadersReceived(details) {
+    let response = {};
+
+    if (details.statusCode === 302 && webRequests.indexOf(details.requestId) < 0) {
+        let matches = details.url.match(/(app|sub)\/(\d+)/);
+
+        if (matches) {
+            let type = matches[1];
+            let id = matches[2];
+
+            switch (SyncedStorage.get('redirect_removed_to')) {
+                case 'community':
+                    response.redirectUrl = `https://steamcommunity.com/${type}/${id}`;
+                    break;
+                case 'steamdb':
+                    response.redirectUrl = `https://steamdb.info/${type}/${id}`;
+                    break;
+                case 'custom':
+                    response.redirectUrl = SyncedStorage.get('redirect_removed_custom')
+                        .replace(/\[TYPE]/, type)
+                        .replace(/\[ID]/, id);
+                    break;
+                default:
+                    break;
+            }
+
+            webRequests.push(details.requestId);
```

Comment:
Isn't that a memory leak here?
If you always push to the array, but never remove anything, it will grow infinitely.

Reply:
I had tried to fix that with `onCompleted`, but it doesn't work well, because we would need permissions for the redirected hosts in order to listen to the completed event for them. So we would need `<all_urls>` permission, and that's a bad idea.

I guess another way to fix it would be to empty the array every once in a while?

Reply:
We only need this for steamcommunity.com as you already said and we have the permission for this.
In order to work properly you'd have to only add to the `Set` if the user chose to redirect to steamcommunity.com